folium/map.py

```diff
@@ -348,144 +350,69 @@ def render(self, **kwargs):
-                                                      "tuple to fields."
-        if all((fields, aliases)):
-            assert isinstance(aliases, (list, tuple))
-            assert len(fields) == len(aliases), "fields and aliases must have" \
-                                                " the same length."
-        assert isinstance(labels, bool), "labels requires a boolean value."
-        assert not all((fields, text)), "Please choose either fields or text."
-        assert any((fields, text)), "Please choose either fields or text."
-        assert isinstance(toLocaleString, bool), "toLocaleString must be " \
-                                                 "boolean."
-        self.valid_kwargs = {"pane": (str,),
-                             "offset": (tuple,),
-                             "direction": (str,),
-                             "permanent": (bool,),
-                             "sticky": (bool,),
-                             "interactive": (bool,),
-                             "opacity": (float, int)}
-        if kwargs:
-            for key in kwargs.keys():
-                assert key in self.valid_kwargs.keys(), \
-                    "The key {} was not available in the keys: {}".format(
-                        key, ', '.join(self.valid_kwargs.keys()))
-                assert isinstance(kwargs[key], self.valid_kwargs[key]), \
-                    "{} must be of the following types: {}".format(
-                        key, self.valid_kwargs[key])
-            self.kwargs = json.dumps(kwargs)
-        self.fields = fields
-        self.aliases = aliases
-        self.text = text.__str__()
-        self.labels = labels
-        self.toLocaleString = toLocaleString
+        self.text = str(text)
+
+        kwargs.update({'sticky': sticky})
+        self.options = self.parse_kwargs(kwargs)
+
         if style:
             assert isinstance(style, str), \
                 "Pass a valid inline HTML style property string to style."
             # noqa outside of type checking.
             self.style = style
 
-
-def create_tooltip(tooltip):
-    """Create a Tooltip object from an unknown input type."""
-    if isinstance(tooltip, Tooltip):
-        return tooltip
-    elif isinstance(tooltip, str):
-        return Tooltip(tooltip)
-    elif isinstance(tooltip, (float, int)):
-        return Tooltip(str(tooltip))
-    else:
-        raise ValueError('Please pass a folium Tooltip object or'
-                         ' a string to the tooltip argument')
+    @staticmethod
+    def parse_kwargs(kwargs):
+        """Validate the provided kwargs and return options as json string."""
+        valid_kwargs = {"pane": (str,),
+                        "offset": (tuple,),
+                        "direction": (str,),
+                        "permanent": (bool,),
+                        "sticky": (bool,),
+                        "interactive": (bool,),
+                        "opacity": (float, int)}
```

Comment:
I've noticed a few additional arguments available in the Leaflet's Tooltip [options](https://leafletjs.com/reference-1.3.2.html#tooltip) that it inherits from `DivOverlay` and `Layer` - `className` and `attribution` that I think would be worthwhile to expose in `**kwargs`.

- `attribution` will pass a string to append to the attribution corner of the window when the tooltip is displayed.
- `className` passes a string to give a class to the Tooltip that could be further used via in an inline `<style></style>` tag in the `Map` template down the road.

Reply:
If we're validating the options, then indeed we should have all available options in there. 

* I moved the dict with valid options to the class attributes since it doesn't change between instances. 
* I also added a function that converts the options to lowerCamelCase. So users can provide the options as Python style variable names and it will still work.